### DISCUSSION TITLE
ESP32-4827S043 board from SUNTON - Flickers

### DISCUSSION THREAD

**Discussion by werde45** [moononournation/Arduino_GFX#291] [General]
Hi,

i´m about to use this display, but everything flickers. I try the simple Clock demo or LVGL Demos. Everything is flickering. When i try the precompiled binary from reseller no flickering at all. I cannot find the cause of it. Even my own project is flickering. I have some of this displays and its always the same. 
When you look at the display from am angle is seems to be an problem of refresh or something.
Any Idea?

**Comment by moononournation** [comment]
Recently, this is the know issue for the ESP32 driving RGB display. Seems idf 5 enhanced so feature to fix this, look forward this will apply to arduino-esp32.

**Reply by werde45** [reply]
Okay. But the demos runs fine on the same display, thus means it is possible to work fine. How to find the problem? The Chinese vendor says it works fine….

**Reply by moononournation** [reply]
Some version of display can run fine in some condition, but I do not have all version in hand.

**Reply by werde45** [reply]
Thats not the point. The same display is able to run fine and the same project compiled by my self is flickering.

**Reply by werde45** [reply]
I´m usind the same versions as the vendor. Now 1.28

**Reply by werde45** [reply]
Arduino_ESP32RGBPanel *bus = new Arduino_ESP32RGBPanel(
    GFX_NOT_DEFINED /* CS */, GFX_NOT_DEFINED /* SCK */, GFX_NOT_DEFINED /* SDA */,
    40 /* DE */, 41 /* VSYNC */, 39 /* HSYNC */, 42 /* PCLK */,
    45 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
    5 /* G0 */, 6 /* G1 */, 7 /* G2 */, 15 /* G3 */, 16 /* G4 */, 4 /* G5 */,
    8 /* B0 */, 3 /* B1 */, 46 /* B2 */, 9 /* B3 */, 1 /* B4 */
);

**Reply by moononournation** [reply]
I did not have this display in hand. You are the second one asking for this display flicking problems. I heard other said LovyanGFX can fix this, please get it a try.

**Reply by werde45** [reply]
Okay, but the seller stated, only your library works and is enough. And I’m not sure, how to include lovyan in your clock demo.

**Comment by h5n1xp** [comment]
> Hi,
> 
> i´m about to use this display, but everything flickers. I try the simple Clock demo or LVGL Demos. Everything is flickering. When i try the precompiled binary from reseller no flickering at all. I cannot find the cause of it. Even my own project is flickering. I have some of this displays and its always the same. When you look at the display from am angle is seems to be an problem of refresh or something. Any Idea?

Have you tried using an Arduino_Canvas? That buffers the output so the screen is updated in a single refresh.

**Comment by caipifrosch** [comment]
> Hi,
> 
> i´m about to use this display, but everything flickers. I try the simple Clock demo or LVGL Demos. Everything is flickering. When i try the precompiled binary from reseller no flickering at all. I cannot find the cause of it. Even my own project is flickering. I have some of this displays and its always the same. When you look at the display from am angle is seems to be an problem of refresh or something. Any Idea?

Same at my end here. Looks like some kind of refresh problem when looking from the side to the display. Tried running in diffent tasks with different priorities and also increased stack size - no change.

**Reply by moononournation** [reply]
This board also support a 800x480 screen and it have better result.

**Reply by werde45** [reply]
Im switcht to the 8048S043 Display and the flicker is gone. But the reseller or manufacturer is not willing to help. Perhabs my chinese it way to bad...

**Reply by caipifrosch** [reply]
> Im switcht to the 8048S043 Display and the flicker is gone. But the reseller or manufacturer is not willing to help. Perhabs my chinese it way to bad...

Good to hear, but I need to go with with 470x270 resolution for my project. My display came with perfectly running demo. So it must be possible without flicker. Probably framework incompatability.

**Reply by werde45** [reply]
Belive my: Me and my friend try for some weeks a lot of different librarys and settings. No success at all. And the Manufacturer is even not helpful. I think that most buy the cheap one or the one with higher resolution and thats why only some complains about the problem. If you ever find a solution, please let us all know. I have two of the displays...but cannot use it. Its not an problem of LVGL.

**Reply by moononournation** [reply]
may be try disable psram

**Comment by Thomas8607** [comment]
My display also flickers, but only for a few seconds after uploading the program.

**Comment by mignon-ne** [comment]
Not sure, if the case is the same.
I also contact vendor and they claimed it is normal for TFT and recommend IPS screen.
By the way, this is very frustrated.
But when I exploring the board seems some user applied difference parameters so I have tried and seems solved.

May be if anyone can reconfirm by trying.
By changing front_porch and pulse_width both hsync and vsync.
Arduino_ESP32RGBPanel *bus = new Arduino_ESP32RGBPanel(
...
    0 /* hsync_polarity */, **8** /* hsync_front_porch */, **4** /* hsync_pulse_width */, 43 /* hsync_back_porch */,
    0 /* vsync_polarity */, **8** /* vsync_front_porch */, **4** /* vsync_pulse_width */, 12 /* vsync_back_porch */,
To be
    0 /* hsync_polarity */, 1 /* hsync_front_porch */, 1 /* hsync_pulse_width */, 43 /* hsync_back_porch */,
    0 /* vsync_polarity */, 3 /* vsync_front_porch */, 1 /* vsync_pulse_width */, 12 /* vsync_back_porch */,

Cheers,

**Reply by moononournation** [reply]
I do not have this display in hand. If someone can confirm it works, I can commit the changes.

**Reply by caipifrosch** [reply]
Big thanks to mignon-ne! 

I can confirm that these settings do work on my board (ESP32-4827S043). 

No intensive testing. Just 5min runtime.

Funny, as new settings violate ILI6485D specs.

**Reply by Xylopyrographer** [reply]
@moononournation  - I'd be glad to chip in to buy a  ESP32-4827S043C for you. How would I go about sending you some cash or have one shipped to you?

**Reply by moononournation** [reply]
If tested ok, please provide a Youtube link of PDQgraphicstest. I will add the dev board declaration to the example.

**Reply by Xylopyrographer** [reply]
Video of the Sunton ESP32-4827S043C running the PDQ Graphics test: [https://youtube.com/shorts/QY09u37htIk?feature=share](https://youtube.com/shorts/QY09u37htIk?feature=share)

Running with the settings supplied by @mignon-ne  above except with the preferred_speed set to 10000000 (10 MHz).

Setting the  preferred_speed set to 12000000 (12 MHz) or to 9000000 (9 MHz) produced lower PDQ scores. Using 10 MHz was the fastest result. No display flickering or tearing seen (using the sync settings above) at either 9, 10, or 12 MHz.

Change the name of the attached file to 4827_PDQgraphicstest.ino as GitHub doesn't allow upload of ".ino" files.
[4827_PDQgraphicstest_ino.txt](https://github.com/moononournation/Arduino_GFX/files/12444505/4827_PDQgraphicstest_ino.txt)

**Comment by werde45** [comment]
Thanks for your help. 
I will try it too, but can't get to it at the moment. 
But what annoys me so much is that I have written to the manufacturer and the seller several times and it always said that there was no problem and also only I have the problem with the display. I have since changed to the IPS display, but that means a lot of adjustment due to the higher resolution. This is certainly not good service from the manufacturer or dealer.

**Comment by Xylopyrographer** [comment]
@werde45

Hello. Just starting to test out this board. Not as yet having much luck with getting the display stable though, as you say, the demo code that it shipped with works like butter.

What is the file name  of the precompiled binary  that you loaded back on to the board? I've not had luck reverting back to the shipped firmware load.

Also wondering if anyone has a part number and/or source for the mating connector (prewired would be great) that matches the three 1.5mm spacing connectors around the outside of the board (SPI, UART1/USB, GND/RXD/TXD/5V)?

Likewise, the female connector that came with the board does not fit the 1mm spacing GND, 3.3V, IO17, IO18 board connector either, so a source for that is needed is anyone would have a source.

Many thanks.

**Reply by caipifrosch** [reply]
AliExpress seller delivered a .zip package with a folder structure. Returning to stock version worked fine when using "TN 4827 CTP-GT911.bin" in "8-burn operation" folder for my capacitive touch version of the board. Used standard esp flash tool for this.

Again, when compiling delivered sources my panel flickered. Worked fine only after using mignon-ne suggested display settings, but these are out of display spec.

**Reply by moononournation** [reply]
mignon-ne？

**Reply by Xylopyrographer** [reply]
Thank-you for the reply. That is the .bin file I was trying to flash. Couldn't get it to run using one tool so reverted to using esptool.py. That worked and it's back to the factory load.

Compiling other examples I ran did require the modified setting as @mignon-ne wrote above to prevent glitching.

I noted that  `TN 4827 CTP-GT911.bin` was built with arduino-esp32 core v2.0.3 which is based on idf-release/v4.4-0-gd3340 and (presuambly) whatever version of Arduino_GFX was current as of October 2022. (Must be a somewhat recent version though as the `.bin` uses the `Arduino_ESP32RGBPanel` class.

My environment uses arduino-esp32 core v2.0.10 (which is based on idf-release/v4.4.5) and  Arduino_GFX v1.3.7. If I was to venture a guess, I'd say something changed in the core and/or the idf-release that affected the timing of calls used by Arduino_GFX. So while it appears that the modified settings violate the ILI6485D specs, in practice it works out that the modified timing is close to the values specified in the original version of the `Arduino_ESP32RGBPanel` bus.

But I could also be entirely wrong 😄.

Big shout out to @moononournation for this library! Many thanks to you. 🙏

**Reply by fvanroie** [reply]
I am using [these settings](https://github.com/HASwitchPlate/openHASP/blob/19ed33433e5b67f54f6a4737be13a837b6b29820/user_setups/esp32s3/sunton-esp32-s3-tft.ini#L68-L78) based on the datasheet and they seem to work fine.

**Reply by Xylopyrographer** [reply]
@fvanroie Thanks for posting.

Ran three tests using the sketch below. Uncomment one section of settings at a time to reproduce.

For reference, development environment is:

**DUT:** Sunton ESP32-4827S043C (capacitive touch screen, ESP32-S3-N16R8).

**Arduino IDE: Version**: 2.1.1, Date: 2023-06-30T16:00:43.829Z, CLI Version: 0.32.3

**Host:** macOS v13.5 (22G74) Ventura

**arduino-esp32 Core:** v2.0.11

**GFX Library for Arduino:** v1.3.7

**From sketch upload:**
    esptool.py v4.5.1
    Chip is ESP32-S3 (revision v0.1)
    Features: WiFi, BLE
    Crystal is 40MHz

**Arduino IDE Settings:**
Board: "ESP323 Dev Module"
USB CDC On Boot: "Disabled"
CPU Frequency: "240MHz (WiFi)"
Core Debug Level: "None"
USB DFU On Boot: "Disabled"
Erase All Flash Before Sketch Upload: "Disabled"
Events Run On: "Core 1"
Flash Mode: "QI0 80MHz"
Flash Size: "16MB (128Mb)"
JTAG Adapter: "Disabled"
Arduino Runs On: "Core 1"
USB Firmware MSC On Boot: "Disabled"
Partition Scheme: "8M with spiffs (3MB APP/1.5MB SPIFFS)"
PSRAM: "OPI PSRAM"
Upload Mode: "UARTO / Hardware CDC"
Upload Speed: "921600"
USB Mode: "Hardware CDC and JAG"

**Results:** See comments in the sketch.

Would be interesting to see what results are obtained by different versions of the arduino-esp32 core.

```
//
// Sunton ESP32-4827S043C Display Settings Test

#include <Arduino_GFX_Library.h>

// SETTING 1 - Original settings for this display in the Arduino_GFX library
//     UNSUCESSFUL: Columns jitter, (columns appear and disappear shifting left and right); display flickering evident
// Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
//     40 /* DE */, 41 /* VSYNC */, 39 /* HSYNC */, 42 /* PCLK */,
//     45 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
//     5 /* G0 */, 6 /* G1 */, 7 /* G2 */, 15 /* G3 */, 16 /* G4 */, 4 /* G5 */,
//     8 /* B0 */, 3 /* B1 */, 46 /* B2 */, 9 /* B3 */, 1 /* B4 */,
//     0 /* hsync_polarity */, 8 /* hsync_front_porch */, 4 /* hsync_pulse_width */, 43 /* hsync_back_porch */,
//     0 /* vsync_polarity */, 8 /* vsync_front_porch */, 4 /* vsync_pulse_width */, 12 /* vsync_back_porch */,
//     1 /* pclk_active_neg */, 9000000 /* prefer_speed */ );

// SETTING 2 - Settings posted by mignon-ne
//    SUCESSFUL: No column jitter; display stable, no flickering.
// Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
//     40 /* DE */, 41 /* VSYNC */, 39 /* HSYNC */, 42 /* PCLK */,
//     45 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
//     5 /* G0 */, 6 /* G1 */, 7 /* G2 */, 15 /* G3 */, 16 /* G4 */, 4 /* G5 */,
//     8 /* B0 */, 3 /* B1 */, 46 /* B2 */, 9 /* B3 */, 1 /* B4 */,
//     0 /* hsync_polarity */, 1 /* hsync_front_porch */, 1 /* hsync_pulse_width */, 43 /* hsync_back_porch */,
//     0 /* vsync_polarity */, 3 /* vsync_front_porch */, 1 /* vsync_pulse_width */, 12 /* vsync_back_porch */,
//     1 /* pclk_active_neg */, 9000000 /* prefer_speed */ );

// SETTING 3 - Settings linked by fvanroie
//    UNSUCESSFUL: Rows jitter, (rows appear and disappear at random; display flickering evident.
// Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
//     40 /* DE */, 41 /* VSYNC */, 39 /* HSYNC */, 42 /* PCLK */,
//     45 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
//     5 /* G0 */, 6 /* G1 */, 7 /* G2 */, 15 /* G3 */, 16 /* G4 */, 4 /* G5 */,
//     8 /* B0 */, 3 /* B1 */, 46 /* B2 */, 9 /* B3 */, 1 /* B4 */,
//     0 /* hsync_polarity */, 64 /* hsync_front_porch */, 4 /* hsync_pulse_width */, 43 /* hsync_back_porch */,
//     0 /* vsync_polarity */, 64 /* vsync_front_porch */, 4 /* vsync_pulse_width */, 12 /* vsync_back_porch */,
//     1 /* pclk_active_neg */, 9000000 /* prefer_speed */ );

Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
    480 /* width */, 272 /* height */, rgbpanel);

#define GFX_BL 2

void setup( void ) {
    gfx->begin();
    gfx->fillScreen( BLACK );

    pinMode( GFX_BL, OUTPUT );
    digitalWrite( GFX_BL, HIGH );

    gfx->setCursor( 10, 10 );
    gfx->setTextColor( RED );
    gfx->println( "Hello World! - Sunton ESP32-4827S043C Display Settings Test" );

    delay( 2000 ); // 2 seconds
}

void loop() {
    gfx->setCursor( random( gfx->width() ), random( gfx->height() ) );
    gfx->setTextColor( random( 0xffff ), random( 0xffff ) );
    gfx->setTextSize( random( 6 ) /* x scale */, random( 6 ) /* y scale */, random( 2 ) /* pixel_margin */ );
    gfx->println( "Hello World!" );

    delay( 500 ); // 1/2 second
}


//  -- EOF --
```

**Comment by Xylopyrographer** [comment]
If of interest, I posted a sketch of a fully functional version of the LVGL_Arduino.ino demo. Link is [here](https://forum.lvgl.io/t/how-to-setup-the-lvgl-library-for-esp32-s3-wroom-1-with-tft-display-800-480/12302/13).

Target is the Sunton ESP32-4827S043C (480 * 272 TFT LED display with capacitive touch) but comments in the sketch are provided to (probably) make it run on the ESP32-8048S043C (pretty much the same board with an 800 * 480 IPS LED panel).

**Reply by moononournation** [reply]
great! please also provide a Youtube link of the full demo.

**Reply by Frtrillo** [reply]
It still have the flickering issue for me, in the 800*480 display. (When using OPI PSRAM)

**Comment by Xylopyrographer** [comment]
Video demo of the ESP32-4327S043C running the LVGL v8 Widgets demo (using the dark theme!) at: [https://youtu.be/FejIzV66-Z0](https://youtu.be/FejIzV66-Z0)
